app/controllers/authors_controller.rb

```diff
@@ -273,7 +273,7 @@ def extension
   end
 
   def update
-    if @author.username != a_params[:username]
+    if a_params[:username] != "" && @author.username != a_params[:username]
```

Comment:
Ruby has a .empty? you can use here instead I believe.

Reply:
nice catch! fixed :)